client/src/hooks/graphql/use-with-data.tsx

```diff
@@ -21,7 +21,7 @@ const initialState: LoadingState = {
 
 interface UpdateFunc<T> {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  callback: (data: T) => Promise<any>;
+  action: (data: T) => Promise<any>;
```

Comment:
why `Promise<any>` instead of `Promise<void>`...something is weird if callers are expecting return objects w this pattern?